project/DottySupport.scala

```diff
@@ -0,0 +1,114 @@
+package scala.build
+
+import sbt._
+import sbt.Keys._
+
+import sbt.librarymanagement.{
+  ivy, DependencyResolution, ScalaModuleInfo, UpdateConfiguration, UnresolvedWarningConfiguration
+}
+
+/** Settings needed to compile with Dotty,
+ *  Only active when sbt is started with `sbt -Dscala.build.compileWithDotty=true`
+ *  This is currently only used to check that the standard library compiles with
+ *  Dotty in .travis.yml.
+ */
+object DottySupport {
+  val dottyVersion = "0.14.0-bin-20190226-afc03c9-NIGHTLY"
+  val compileWithDotty: Boolean =
+    Option(System.getProperty("scala.build.compileWithDotty")).map(_.toBoolean).getOrElse(false)
```

Comment:
That is actually not equivalent, because `s.toBoolean` is not equivalent to `j.l.Boolean.parseBoolean(s)` - `toBoolean` allows only `"true"` and `"false"` (case insensitive), while `parseBoolean` permits all strings.

```scala
scala> java.lang.Boolean.parseBoolean("foo")
res4: Boolean = false

scala> "foo".toBoolean
java.lang.IllegalArgumentException: For input string: "foo"
  at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:327)
  at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)
  at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)
  at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)
  ... 28 elided
```

Additionally, I object to the use of `j.l.Boolean.getBoolean` because it is confusing and an API that does not belong in `j.l.Boolean` (why should `j.l.Boolean` know about system properties??).

Reply:
Agreed, let's keep as-is